shared/domains/chelonia/chelonia.js

```diff
@@ -1098,7 +1099,10 @@ export default (sbp('sbp/selectors/register', {
       if (done) return state
       const stateCopy = cloneDeep(state)
       try {
-        await sbp('chelonia/private/in/processMessage', GIMessage.deserialize(event, this.transientSecretKeys, state), state)
+        await sbp('chelonia/private/in/processMessage', GIMessage.deserialize(event, this.transientSecretKeys, state), state, undefined, contractName)
+        if (!contractName && state._vm) {
+          contractName = state._vm?.type
```

Comment:
You've already verified `state._vm` exists in the `if`, so this `state.vm?.` seems unnecessary AFAICT.

You can either remove the check in the `if`, or remove the `?` here.